rest-common/src/main/java/uk/ac/ebi/quickgo/rest/search/query/SortedSolrQuerySerializer.java

```
@@ -0,0 +1,67 @@
+package uk.ac.ebi.quickgo.rest.search.query;
+
+import uk.ac.ebi.quickgo.rest.search.SolrQueryStringSanitizer;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static uk.ac.ebi.quickgo.rest.search.query.SolrQueryConverter.CROSS_CORE_JOIN_SYNTAX;
+import static uk.ac.ebi.quickgo.rest.search.query.SolrQueryConverter.SOLR_FIELD_SEPARATOR;
+
+/**
+ * <p>This class defines an algorithm for serializing {@link QuickGOQuery}s into a corresponding
+ * Solr query String. The implementation creates a corresponding String query that returns
+ * results in the expected order of relevance, where Solr scoring is performed on all subclasses
+ * of {@link QuickGOQuery}.
+ *
+ * Created 02/08/16
+ * @author Edd
+ */
+public class SortedSolrQuerySerializer implements QueryVisitor<String> {
+    private final SolrQueryStringSanitizer queryStringSanitizer;
+
+    public SortedSolrQuerySerializer() {
+        this.queryStringSanitizer = new SolrQueryStringSanitizer();
```

even if it's being initialised in this method? (so cannot be null, unless the `new` keyword, and if initialisation failed, java should throw an exception)

```java
this.queryStringSanitizer = new SolrQueryStringSanitizer();
```